cmd/node/factory/structs.go

```diff
@@ -1323,6 +1332,13 @@ func newShardBlockProcessor(
 		return nil, errors.New("could not create transaction statisticsProcessor: " + err.Error())
 	}
 
+	txSimulatorProcessorArgs := argsNewTxProcessor
+	txSimulatorProcessorArgs.Accounts = disabled.NewAccountsAdapter()
```

Comment:
instead of disabled accounts adapter - you have to create a read only accounts adapter. thus you need a new component which gets the accountsDB as parameter and does all the reading from it, and in case of writes it writes into a cache, or it has an empty implementation

Reply:
done